gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/resources/openapi/openapi-environments.yaml

```diff
@@ -1642,6 +1669,81 @@ components:
             published:
               type: boolean
               description: Whether the page is published or not
+        PortalArea:
+          type: string
+          description: The portal area (used by portal navigation items)
+          enum: [HOMEPAGE, TOP_NAVBAR]
+        BasePortalNavigationItem:
+          type: object
+          description: Base portal navigation item
+          properties:
+            id:
+              type: string
+              description: The unique ID of the navigation item
+            organizationId:
+              type: string
+              description: The organization ID
+            environmentId:
+              type: string
+              description: The environment ID
+            title:
+              type: string
+              description: The title of the navigation item
+            type:
+              type: string
+              description: The type of the navigation item
+              enum: [PAGE, FOLDER, LINK]
+            area:
+              $ref: "#/components/schemas/PortalArea"
+            parentId:
+              type: string
+              description: The parent ID of the navigation item
+            order:
+              type: integer
+              description: The order of the navigation item
+          required: [id, organizationId, environmentId, title, type, area, order]
+          discriminator:
+            propertyName: type
+            mapping:
+              PAGE: "#/components/schemas/PortalNavigationPage"
+              FOLDER: "#/components/schemas/PortalNavigationFolder"
+              LINK: "#/components/schemas/PortalNavigationLink"
+        PortalNavigationPage:
+          allOf:
+            - $ref: "#/components/schemas/BasePortalNavigationItem"
+            - properties:
+                configuration:
+                  type: object
+                  properties:
+                    portalPageContentId:
+                      type: string
+                      description: The UUID of the portal page content
+                  required: [portalPageContentId]
+        PortalNavigationFolder:
+          allOf:
+            - $ref: "#/components/schemas/BasePortalNavigationItem"
```

Comment:
Shoudln't we add here the possibility for a folder to have "children" of type PortalNavigationItem ?
Otherwise, we won't be able to return a structured list of pages. Everything will be flat.